proof/crefine/RISCV64/IsolatedThreadAction.thy

```diff
@@ -982,24 +918,38 @@ lemma tcbSchedEnqueue_tcbPriority[wp]:
   apply (wp | simp cong: if_cong)+
   done
 
-crunch obj_at_prio[wp]: cteDeleteOne "obj_at' (\<lambda>tcb. P (tcbPriority tcb)) t"
-  (wp: crunch_wps setEndpoint_obj_at'_tcb
-       setThreadState_obj_at_unchanged setNotification_tcb setBoundNotification_obj_at_unchanged
-        simp: crunch_simps unless_def)
-
-lemma setThreadState_no_sch_change:
-  "\<lbrace>\<lambda>s. P (ksSchedulerAction s) \<and> (runnable' st \<or> t \<noteq> ksCurThread s)\<rbrace>
-      setThreadState st t
-   \<lbrace>\<lambda>rv s. P (ksSchedulerAction s)\<rbrace>"
-  (is "NonDetMonad.valid ?P ?f ?Q")
-  apply (simp add: setThreadState_def setSchedulerAction_def)
-  apply (wp hoare_pre_cont[where a=rescheduleRequired])
-  apply (rule_tac Q="\<lambda>_. ?P and st_tcb_at' ((=) st) t" in hoare_post_imp)
-   apply (clarsimp split: if_split)
-   apply (clarsimp simp: obj_at'_def st_tcb_at'_def projectKOs)
-  apply (wp threadSet_pred_tcb_at_state)
-  apply simp
-  done
+crunches unbindMaybeNotification, blockedCancelIPC, replyRemoveTCB, cancelSignal
+  for tcbPriority_obj_at'[wp]: "obj_at' (\<lambda>tcb. P (tcbPriority tcb)) t'"
```

Comment:
It would be nice to move these new hoare triple lemmas about the design spec to `Move_C`. We're not completely consistent about this, but the intention would be to move them to Refine (and eventually maybe [DInvs](https://sel4.atlassian.net/browse/VER-1299)).

Reply:
I've added a `FIXME RT: move` for this and mentioned DInvs